### PR TITLE
Pretty parse

### PR DESCRIPTION
**Pull request by aslakhellesoy**
## Summary

Move `pretty` and `AstWalker` to the `gherkin` library

## Motivation and Context

We want to use `pretty` without `@cucumber/react`. Initially in the new example mapping tool (for exporting gherkin from an example map), later possibly in other tools, like a linter.

## How Has This Been Tested?

With automated tests

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue).
- [x] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).

## Checklist:

- [ ] The change has been ported to Java.
- [ ] The change has been ported to Ruby.
- [x] The change has been ported to JavaScript.
- [ ] The change has been ported to Go.
- [ ] The change has been ported to .NET.
- [ ] I've added tests for my code.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have updated the CHANGELOG accordingly.

### CODE REVIEW

**Comment by vincent-psarga** [comment]
Nice :)

That said, why did you remove the description output ? It's pretty useful to have an equivalent output from the source.

Also one thing we should handle too is the tags of the features, scenarios etc.

File: gherkin/javascript/src/pretty/pretty.ts [hunk 1]
```diff
@@ -5,20 +5,13 @@ export default function pretty(
 ): string {
   const feature = gherkinDocument.feature
   let s = feature.keyword + ': ' + feature.name + '\n'
-  if (feature.description) {
```
Comment (vincent-psarga) [inline gherkin/javascript/src/pretty/pretty.ts:8]: we should keep the description.

Some search match are done in the description and the tests for filtering should show where the hit was done.

**Comment by aslakhellesoy** [comment]
I see you’ve made aome improvements to `pretty` in the branch that adds tag search (very cool, how does it perform on large docs?)

We’ll have to carry your changes over when we merge it all.

**Comment by vincent-psarga** [comment]
I can make the changes in your branch too, that should not be much of an issue.

By the way there's something weird in this PR, it looks like you're making the PR to be merged into `react-add-search-bar` and not into `master`. I don't know when `react-add-search-bar` will be really ready to be merged into master, I think I'd be better to have this PR directly sent to `master` (or there's something not correctly displaying on GitHub)

**Comment by vincent-psarga** [comment]
Just a thing that popped in my mind: if the `AstWalker` is moved to `@cucumber/gherkin`, that means we'll need to use the library as a dependency instead of a devDependency in react, which I guess wouod make the lib heavier :/
Maybe we coyld extract that to akother lib that only requires `@cucumber/messages` ?

**Comment by aslakhellesoy** [comment]
Sure, we can create a new library, but before we do I'd like to understand the implications of keeping it where it is now (in gherkin).

Can we quantify what problems we might run into by keeping it in gherkin? I'm not worried about size, I'm more worried about frequent releases.

**Comment by vincent-psarga** [comment]
I would worry about size, we already have warnings from `es-slint` about the size of the packaged dist.
It's quite heavy for a simple HTML report, and what's without the messages that end up in the HTML report once everything is there.

Just as an example, when running cucumber ruby suite:
```shell
 ls -lah | grep report                            
-rw-r--r--    1 vincent.pretre  staff   3.8M Jun  5 10:07 report.html
-rw-r--r--    1 vincent.pretre  staff   2.3M Jun  5 15:18 report.messages```

And that's without Gherkin, ElasticLunr etc embedded in the HTML page. The messages are pretty heavy already, let's try not to make this worst.

**Pull request merged**